### PR TITLE
[ast] Refactor Builtin printing logic from ASTPrinter -> BuiltinInst.

### PR DESCRIPTION
**Pull request by gottesmm**
This is so I can use this functionality later to stringly convert polymorphic builtins into other statically overloaded builtins by munging names.

In terms of the ASTPrinter, I rewrote it to use the BuiltinInst part.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: include/swift/Strings.h [hunk 1]
```diff
@@ -40,46 +40,76 @@ constexpr static const char MANGLING_MODULE_CLANG_IMPORTER[] =
...
 /// The name of the Builtin type for Vector
 constexpr static const char BUILTIN_TYPE_NAME_VEC[] = "Builtin.Vec";
+constexpr static const char BUILTIN_TYPE_NAME_VEC_NO_NAMESPACE[] = "Vec";
 /// The name of the Builtin type for SILToken
 constexpr static const char BUILTIN_TYPE_NAME_SILTOKEN[] = "Builtin.SILToken";
+constexpr static const char BUILTIN_TYPE_NAME_SILTOKEN_NO_NAMESPACE[] =
+    "SILToken";
 /// The name of the Builtin type for Word
 constexpr static const char BUILTIN_TYPE_NAME_WORD[] = "Builtin.Word";
+constexpr static const char BUILTIN_TYPE_NAME_WORD_NO_NAMESPACE[] = "Word";
```
Comment (jrose-apple) [inline include/swift/Strings.h]: Can you clean up the duplication here? Either by removing the original constants, or by not adding the new ones and instead adding a fixed offset (`llvm::array_lengthof(BUILTIN_TYPE_NAME_PREFIX)`).

Comment (gottesmm) [reply include/swift/Strings.h]: I think that would require us to change these to StringLiteral or the like. Last time I did that I ran into problems. Maybe now that works now. Let me try it.

Comment (jrose-apple) [reply include/swift/Strings.h]: Hm, yeah, I see. StringLiteral doesn't have a constructor that would accept an offset. But maybe not everywhere needs to be using StringLiteral either—it's only *really* important for static const / constexpr data.

Comment (gottesmm) [reply include/swift/Strings.h]: I figured out how to do this using a composition class with StringLiteral. It was pretty difficult to find the exact incantation. Turns out the trick to make it work is to use `{"$STRING"}` constructor. Then in my composition class I have a method that returns a StringRef that is the StringLiteral.drop_front(NUM_CHARS).

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: lib/SwiftDemangle/MangleHack.cpp [hunk 2]
```diff
@@ -56,7 +56,7 @@ _swift_mangleSimpleClass(const char *module, const char *class_) {
   size_t moduleLength = strlen(module);
   size_t classLength = strlen(class_);
   char *value = nullptr;
-  if (strcmp(module, swift::STDLIB_NAME) == 0) {
+  if (swift::STDLIB_NAME.compare(module) == 0) {
```
Comment (jrose-apple) [inline lib/SwiftDemangle/MangleHack.cpp]: These two are just checking for equality, so no need to use `compare` now that these are StringLiterals.

**Review by jrose-apple — COMMENTED** [review]
Thank you for the name cleanup!

File: lib/SwiftDemangle/MangleHack.cpp (hunk 2 above)
Comment (gottesmm) [reply lib/SwiftDemangle/MangleHack.cpp]: I am going to do that in a follow up commit. I am not trying to change anything beyond surface level here.

**Pull request merged**